module.js

```diff
@@ -217,19 +211,25 @@ instance.prototype.GetPlaybackState = function(){
             self.checkFeedbacks('is-shuffle');
         }
 
-        let songProgress = data.body.progress_ms;
-        let songDuration = data.body.item.duration_ms
-        let songPercentage = songProgress/songDuration;
-        
-        songPercentage = songPercentage*100;
-        songPercentage = songPercentage.toFixed(0);
-
-        songProgress = songProgress/1000;
-        songProgress = songProgress.toFixed(0);
-
-        songDuration = songDuration/1000;
-        songDuration = songDuration.toFixed(0);
+        if (data.body.item.duration_ms) {
+            var songProgress = data.body.progress_ms;
+            var songDuration = data.body.item.duration_ms;
+            var songPercentage = songProgress/songDuration;
 
+            songPercentage = songPercentage*100;
+            songPercentage = songPercentage.toFixed(0);
+    
+            songProgress = songProgress/1000;
+            songProgress = songProgress.toFixed(0);
+    
+            songDuration = songDuration/1000;
+            songDuration = songDuration.toFixed(0);
+        } else {
+            var songProgress = 0;
```

Comment:
you are going to have a scope issue here. these variables will only exist inside this else block, and will be undefined by the time you are calling `self.setVariable(...` with them

Reply:
Fixed this by moving the variable definitions outside of the loop, also noticed that some other variables would have had the same issue and gave them defaults for when data.body.item is empty.